cmd/compact.go

```diff
@@ -40,14 +40,20 @@ func NewCompactCommand(ctx context.Context) *cobra.Command {
 			- Save the snapshot
 			*/
 			logger := logrus.New()
+			if err := opts.validate(); err != nil {
+				logger.Fatalf("failed to validate the options: %v", err)
+				return
+			}
 
 			options, store, err := BuildRestoreOptionsAndStore(opts.restorerOptions)
 			if err != nil {
 				return
 			}
 
 			cp := compactor.NewCompactor(store, logrus.NewEntry(logger))
-			snapshot, err := cp.Compact(options, opts.needDefragmentation)
+			opts.compactorConfig.RestoreOptions = options
+
+			snapshot, err := cp.Compact(opts.compactorConfig)
```

Comment:
The `CompactorConfig` is used for two different purposes here.
1. To act as the receiver for command-line flags.
2. To pass the above command-line flags as well as restoration options to `cp.Compact()`.

This is a bit confusing. Especially, when restorer uses two different structures for similar two purposes.
1. [RestorerConfig](https://github.com/gardener/etcd-backup-restore/blob/7a4872e4e567a53039402174c0cfdecd7feec1dc/pkg/types/restorer.go#L54-L68) for receiving command-line flags.
2. [RestoreOptions](https://github.com/gardener/etcd-backup-restore/blob/7a4872e4e567a53039402174c0cfdecd7feec1dc/pkg/types/restorer.go#L43-L50) for the `Restore()` function.

I think this distinction makes sense and we should follow that convention for compaction config and compaction options.

Reply:
```go
type CompactOptions struct {
     *RestoreOptions
     *CompactorConfig
}
```

```go
type CompactorConfig struct {
	NeedDefragmentation bool              `json:"needDefrag,omitempty"`
	SnapshotTimeout     wrappers.Duration `json:"snapshotTimeout,omitempty"`
	DefragTimeout       wrappers.Duration `json:"defragTimeout,omitempty"`
}
```
Is this make sense?

Reply:
Can you now take a look?
One doubt, should I move [this](https://github.com/gardener/etcd-backup-restore/pull/361/files#diff-1594bdd07fc72c6a4a7bd517239dff6e4734ac1d236b9726decf318916dca312R55-R58) to a separate function call.

Reply:
I think, it's f9

Reply:
Thanks. Looks good 👍